src/frontend/src/optimizer/plan_node/logical_iceberg_scan.rs

```diff
@@ -37,18 +38,23 @@ use crate::utils::{ColIndexMapping, Condition};
 pub struct LogicalIcebergScan {
     pub base: PlanBase<Logical>,
     pub core: generic::Source,
+    iceberg_scan_type: IcebergScanType,
 }
 
 impl LogicalIcebergScan {
-    pub fn new(logical_source: &LogicalSource) -> Self {
+    pub fn new(logical_source: &LogicalSource, iceberg_scan_type: IcebergScanType) -> Self {
```

Comment:
In my mind, different iceberg scan types should have different schemas. For example, equality delete should contain the exact delete columns that appear in the files instead of all columns of the iceberg table. Maybe we should change the `core` to have a `generic::IcebergScan` instead of `generic::Source`